talosctl validate panics

## Bug Report

### Description

```
./talosctl validate -c XXX.yaml -m metal

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x13eeced]

goroutine 1 [running]:
github.com/talos-systems/talos/pkg/machinery/config/types/v1alpha1.(*Config).Validate(0xc0008cf6b0, 0x2d4e8c0, 0x3a0e170, 0x0, 0x0)
	/src/pkg/machinery/config/types/v1alpha1/v1alpha1_validation.go:78 +0xcd
github.com/talos-systems/talos/cmd/talosctl/cmd/mgmt.glob..func8(0x3a6d6c0, 0xc00019b580, 0x0, 0x4, 0x0, 0x0)
	/src/cmd/talosctl/cmd/mgmt/validate.go:38 +0xdc
github.com/spf13/cobra.(*Command).execute(0x3a6d6c0, 0xc00019b480, 0x4, 0x4, 0x3a6d6c0, 0xc00019b480)
	/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:852 +0x47c
github.com/spf13/cobra.(*Command).ExecuteC(0x3a6ca40, 0x3a8c8a0, 0x2ae6366, 0x9)
	/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:960 +0x375
github.com/talos-systems/talos/cmd/talosctl/cmd.Execute(0x0, 0x0)
	/src/cmd/talosctl/cmd/root.go:42 +0x235
main.main()
	/src/cmd/talosctl/main.go:18 +0x33
```

That's 0.9.0 release.

As a side not, I think we should fuzz our config parser / validator.

--- a/pkg/machinery/config/types/v1alpha1/v1alpha1_validation.go
+++ b/pkg/machinery/config/types/v1alpha1/v1alpha1_validation.go
@@ -73,24 +73,24 @@ func (c *Config) Validate(mode config.RuntimeMode, options ...config.ValidationO
 	if mode.RequiresInstall() {
 		if c.MachineConfig.MachineInstall == nil {
 			result = multierror.Append(result, fmt.Errorf("install instructions are required in %q mode", mode))
-		}
-
-		if opts.Local {
-			if c.MachineConfig.MachineInstall.InstallDisk == "" && len(c.MachineConfig.MachineInstall.DiskMatchers()) == 0 {
-				result = multierror.Append(result, fmt.Errorf("either install disk or diskSelector should be defined"))
-			}
 		} else {
-			disk, err := c.MachineConfig.MachineInstall.Disk()
-
-			if err != nil {
-				result = multierror.Append(result, err)
-			} else {
-				if disk == "" {
-					result = multierror.Append(result, fmt.Errorf("an install disk is required in %q mode", mode))
+			if opts.Local {
+				if c.MachineConfig.MachineInstall.InstallDisk == "" && len(c.MachineConfig.MachineInstall.DiskMatchers()) == 0 {
+					result = multierror.Append(result, fmt.Errorf("either install disk or diskSelector should be defined"))
 				}
+			} else {
+				disk, err := c.MachineConfig.MachineInstall.Disk()
 
-				if _, err := os.Stat(disk); os.IsNotExist(err) {
-					result = multierror.Append(result, fmt.Errorf("specified install disk does not exist: %q", c.MachineConfig.MachineInstall.InstallDisk))
+				if err != nil {
+					result = multierror.Append(result, err)
+				} else {
+					if disk == "" {
+						result = multierror.Append(result, fmt.Errorf("an install disk is required in %q mode", mode))
+					}
+
+					if _, err := os.Stat(disk); os.IsNotExist(err) {
+						result = multierror.Append(result, fmt.Errorf("specified install disk does not exist: %q", c.MachineConfig.MachineInstall.InstallDisk))
+					}
 				}
 			}
 		}

